.github/workflows/manual-package.yml

```diff
@@ -0,0 +1,98 @@
+name: Manual package creation
+on:
+  workflow_dispatch:
+    inputs:
+      skipAtomicE2E:
+        description: 'Skip Atomic end to end tests'
+        type: boolean
+        required: false
+      skipQuanticE2E:
+        description: 'Skip Quantic end to end tests'
+        type: boolean
+        required: false
+      skipUnitTests:
+        description: 'Skip Unit tests'
+        type: boolean
+        required: false
+jobs:
+  build:
+    name: 'Build'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/setup
+  lint-check:
+    name: 'Check with linter'
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/setup
+      - run: npm run lint:check
+  unit-test:
+    name: 'Run unit tests'
+    if: ${{ github.event.inputs.skipUnitTests != 'true' }}
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/setup
+      - run: npx --no-install lerna run copy:staticresources --scope @coveo/quantic
+      - run: npm test
+  e2e-atomic-test:
+    name: 'Run e2e tests on Atomic'
+    if: ${{ github.event.inputs.skipAtomicE2E != 'true' }}
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/setup
+      - uses: ./.github/actions/e2e-atomic
+        with:
+          cypress-record-key: ${{ secrets.ATOMIC_CYPRESS_RECORD_KEY }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+  e2e-quantic-test:
+    name: 'Run e2e tests on Quantic'
+    if: ${{ github.event.inputs.skipQuanticE2E != 'true' }}
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/setup
+      - uses: ./.github/actions/e2e-quantic
+        with:
+          cypress-record-key: ${{ secrets.QUANTIC_CYPRESS_RECORD_KEY }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          clientid: ${{ secrets.SFDX_AUTH_CLIENT_ID }}
+          jwtkey: ${{ secrets.SFDX_AUTH_JWT_KEY }}
+  bump-version:
+    name: 'Bump version'
+    needs:
+      - 'lint-check'
+      - 'unit-test'
+      - 'e2e-atomic-test'
+      - 'e2e-quantic-test'
+    runs-on: ubuntu-latest
+    if: always()
```

Comment:
You cannot have conditional `needs`, so instead, I went with this approach: 

 * `if: always()` so that `bump-version` is always ran, even if a part previous job is skipped.
 * Add steps that verify that each `needs` was not a failure.
 * Net net is that you can skip a job (end to end test) and still bump a version and start a release, but if a job is not skipped, it must be a success.